src/plugins/workspace/server/saved_objects/workspace_saved_objects_client_wrapper.ts

```diff
@@ -272,6 +281,10 @@ export class WorkspaceSavedObjectsClientWrapper {
       objects: Array<SavedObjectsBulkCreateObject<T>>,
       options: SavedObjectsCreateOptions = {}
     ): Promise<SavedObjectsBulkResponse<T>> => {
+      // Only OSD admin can bulkCreate workspace
+      if (objects.some((obj) => obj.type === WORKSPACE_TYPE) && !options?.overwrite)
```

Comment:
Why we need add `options.overwrite` flag check here? As my understanding, user can create new saved objects when `options.overwrite` equal true. See https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/core/server/saved_objects/service/lib/repository.ts#L361 Seems `options.overwrite` only worked when saved object id exists. Shall we handle this case?

Reply:
In create method, I will add the check `options.id` and `options.overwrite` to make users with admin permission update workspace successfully. In bulkCreate method, I will not add the check as the bulkCreate method is not called in the workspaceClient. Is it feasible to do this?

Reply:
I'd recommend to do these. I think object.id will be stored in objects. I think you can have a test in your local. You can try to pass a workspace saved objects array without id and `options.overwrite` equal `true`. Then the  workspace can be created even not an OSD admin.  I think it will bypass the OSD admin permission check.

Reply:
In my latest commit, if users without OSD admin permission want to bulkCreate workspace, they will fail. 
```
wrapper.bulkCreate([{ type: 'workspace', attributes: {} }])
```
If users without OSD admin permission want to create a workspace, they will fail.
```
wrapper.create('workspace', {name: 'test'}, {})
```
If users with admin permission want to update a workspace, they will success.
```
wrapper.create('workspace', {name: 'test'}, {id: 'ws', overwrite: true})
```

Reply:
I think we can safely say user is trying to `update a workspace` only if:
1. Using overwrite mode and request with object id.
3. The workspace user wants to overwrite indeed exists.